cr/state/proposalmanager.go

```diff
@@ -193,6 +193,20 @@ func (p *ProposalManager) updateProposals(height uint32,
 					unusedAmount += getProposalTotalBudgetAmount(v.Proposal)
 					continue
 				}
+				if v.Proposal.ProposalType == payload.ChangeProposalOwner {
+					proposal := p.getProposal(v.Proposal.PreviousHash)
+					originRecipient := proposal.Proposal.Recipient
+					emptyUint168 := common.Uint168{}
+					p.history.Append(height, func() {
+						proposal.ProposalOwner = v.Proposal.NewOwnerPublicKey
+						if v.Proposal.Recipient != emptyUint168 {
+							proposal.Proposal.Recipient = v.Proposal.Recipient
```

Comment:
need to reset recipient as ProposalOwner
1.add 'Recpient' field into ProposalState
2.set value of ProposalState Recipient when registering proposal
3.reset value of ProposalState Recipient when changing proposal owner
4.adjust all places where this field is used in check transactions functions，modify from 'proposalState.proposal.Recipient' to 'proposalState.Recipient'